packages/flutter/lib/src/widgets/app.dart

```diff
@@ -148,10 +148,14 @@ class WidgetsApp extends StatefulWidget {
          'must have their initial values '
          '(null, null, and the empty list, respectively).'
        ),
-       assert(onGenerateRoute != null || pageRouteBuilder != null,
-         'If onGenerateRoute is not provided, the pageRouteBuilder must be specified '
-         'so that the default handler will know what kind of PageRoute transition '
-         'bo build.'),
+       assert(
+         builder != null ||
+         onGenerateRoute != null ||
+         pageRouteBuilder != null,
+         'If neither builder nor onGenerateRoute are provided, the '
+         'pageRouteBuilder must be specified so that the default handler '
+         'will know what kind of PageRoute transition to build.'
```

Comment:
I'm not sure if it makes sense to put the emphasis in this message on pageRouteBuilder.  I feel like this could prod somebody in the wrong direction if they actually need builder, or onGenerateRoute.  Is it sufficient to just say "builder, onGenerateRoute, or pageRouteBuilder must be specified."? (Although I guess that pretty much just restates the assert expression)

Reply:
There are other asserts covering various scenarios around needing a builder vs home/routes/onGenerateRoute/etc.  This one in particular will fire after those, and at this point should only fire for someone who has specified `home` or `routes` but not `builder`, `onGenerateRoute`, or `pageRouteBuilder`.

Reply:
Awesome!  Thanks for the quick reply.